### ISSUE TITLE
failovermethod=priority is not supported by dnf (default package manager in Centos 8+ and derivates)

### ISSUE THREAD

**Issue by jozefmares** [saltstack/salt#61427]
**Description**
When installing Saltstack following instruction from https://repo.saltproject.io/#rhel failovermethod=priority is added to repository file (/etc/yum.repos.d/salt-py3-latest.repo). However this configuration value is not supported by default package manager DNF thus it generates warning.

**Setup**
Create new VM, I used RockyLinux.

**Steps to Reproduce the behavior**
1. create new VM, CentOS 8, RockyLinux etc.
2. install Salt following instructions here: https://repo.saltproject.io/#rhel
3. call dnf update
4. you will see error: Invalid configuration value: failovermethod=priority in /etc/yum.repos.d/salt-py3-latest.repo; Configuration: OptionBinding with id "failovermethod" does not exist

**Expected behavior**
No error is shown. Error can fixed by removing failovermethod=priority from repofile.

**Additional context**

**Comment by dmurphy18** [comment]
@waynew The repo files are handled by the SRE Team, see create_repo_files.sh
Will leave this to you if you want to re-assign this to the SRE Team

**Comment by waynew** [comment]
@krionbsd if you could either take a look at this or re-assign to someone else in the realm :+1:

**Comment by krionbsd** [comment]
```
create new VM, CentOS 8, RockyLinux etc.
install Salt following instructions here: https://repo.saltproject.io/#rhel
call dnf update
you will see error: Invalid configuration value: failovermethod=priority in /etc/yum.repos.d/salt-py3-latest.repo; Configuration: OptionBinding with id "failovermethod" does not exist
```
I followed the steps you described, but couldn't reproduce it, did you miss something here?

**Comment by krionbsd** [comment]
> If you look for example to Redhat bugzilla you can see that configuration value is clearly invalid:

right, it's invalid for dnf, but it's valid for yum, I'll look what we might do with it

**Comment by jozefmares** [comment]
I logged in production just to test :)

```
cat /etc/redhat-release
Rocky Linux release 8.5 (Green Obsidian)
```

```
yum update
Invalid configuration value: failovermethod=priority in /etc/yum.repos.d/salt-py3-latest.repo; Configuration: OptionBinding with id "failovermethod" does not exist
```

It seems like DNF is in fact next release of YUM:
https://man7.org/linux/man-pages/man1/yum-aliases.1.html

I guess this part of manual is root cause of our problem:

**It roughly maintains CLI compatibility with YUM** :)

**Comment by dmurphy18** [comment]
@jozefmares Yeah, yum is becoming just a symbolic link to dnf, or more accurately yum and dnf are symbolic links to dnf-3

**Comment by krionbsd** [comment]
@dmurphy18 yup, in this case we might get rid of `failovermethod` for next release

**Comment by dmurphy18** [comment]
Given it is packaging and hence assume you mean next any release, and not till the next major release

**Comment by alan-cugler** [comment]
Ran into the same error above in a test deployment of vRASSC on aws using Centos 8. Though not needed, I can confirm commenting out the offending configuration fixes the warning/issue.

**Comment by dmurphy18** [comment]
Have MR to fix in https://gitlab.com/saltstack/open/qa-master-scripts/-/merge_requests/21

**Comment by dmurphy18** [comment]
MR merged
